### ISSUE TITLE
Mosquitto not accesible from other devices in local network

### ISSUE THREAD

**Issue by SabelOs** [home-assistant/addons#3080]
### Describe the issue you are experiencing

I am trying to access the mosquitto broker addon installed on my raspberry pi running Home Assitant OS, from other devices on my local network. I've read in other discussions, that one needs to tell the mosquitto addon explicitly to be accesible from the local network. When I try to change the  configuration in mosquitto.conf located in /share/mosquitto to have the `listener 1883 0.0.0.0` line and `customize:
  active: true`  the addon won't start. It states `Error: Address already in use`.

I allready tried to include a `mqqt:` section in the main configuration.yaml of Home Assistant but it won't let me save the configuration

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

Almond

### What is the version of the add-on?

6.2.1

### Steps to reproduce the issue

1. Create mosquitto.conf in /sahre/mosquitto, with `listener 1883 0.0.0.0`
2. Change Addon configuration to  
`customize:`
  `active: true`
  `folder: mosquitto`
4. Restart Addon
5. Log:  `Opening ipv4 listen socket on port 1883.`
`Error: Address already in use`

## System Information

version | core-2023.5.4
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.10.11
os_name | Linux
os_version | 6.1.21-v8
arch | aarch64
timezone | Europe/Berlin
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.32.1
Stage | running
Available Repositories | 1283
Downloaded Repositories | 5

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 10.2
-- | --
update_channel | stable
supervisor_version | supervisor-2023.04.1
agent_version | 1.5.1
docker_version | 23.0.6
disk_total | 28.5 GB
disk_used | 5.3 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | Mosquitto broker (6.2.1), Zigbee2MQTT (1.30.4-1), Studio Code Server (5.6.1), Terminal & SSH (9.7.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 1
-- | --
resources | 4
views | 1
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 30. Mai 2023 um 17:52
-- | --
current_recorder_run | 4. Juni 2023 um 18:53
estimated_db_size | 8.30 MiB
database_engine | sqlite
database_version | 3.40.1

</details>

### Anything in the Supervisor logs that might be useful for us?

```txt
Logger: homeassistant.components.hassio
Source: components/hassio/websocket_api.py:125
Integration: Home Assistant Supervisor (documentation, issues)
First occurred: 4. Juni 2023 um 21:59:54 (8 occurrences)
Last logged: 08:40:03

Failed to to call /addons/core_mosquitto/stats - Container addon_core_mosquitto is not running
```

### Anything in the add-on logs that might be useful for us?

```txt
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/mosquitto.sh
[08:27:39] INFO: Setting up user mqtt-user
[08:27:40] INFO: SSL is not enabled
cont-init: info: /etc/cont-init.d/mosquitto.sh exited 0
cont-init: info: running /etc/cont-init.d/nginx.sh
cont-init: info: /etc/cont-init.d/nginx.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun mosquitto (no readiness notification)
services-up: info: copying legacy longrun nginx (no readiness notification)
s6-rc: info: service legacy-services successfully started
[08:27:41] INFO: Starting NGINX for authentication handling...
[08:27:41] INFO: Starting mosquitto MQTT broker...
1685946461: Loading config file /share/mosquitto/mosquitto.conf
2023-06-05 08:27:41: Warning: Mosquitto should not be run as root/administrator.
2023-06-05 08:27:41: mosquitto version 2.0.15 starting
2023-06-05 08:27:41: Config loaded from /etc/mosquitto/mosquitto.conf.
2023-06-05 08:27:41: Loading plugin: /usr/share/mosquitto/go-auth.so
2023-06-05 08:27:41:  ├── Username/password checking enabled.
2023-06-05 08:27:41:  ├── TLS-PSK checking enabled.
2023-06-05 08:27:41:  └── Extended authentication not enabled.
2023-06-05 08:27:41: Opening ipv4 listen socket on port 1883.
2023-06-05 08:27:41: Opening ipv4 listen socket on port 1883.
2023-06-05 08:27:41: Error: Address already in use
[06:27:41] WARNING: Halt add-on
s6-rc: info: service legacy-services: stopping
[06:27:41] INFO: Service restart after closing
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

### Additional information

Please excuse any mistakes, I am by no means an expert in this topic.

**Comment by agners** [comment]
> I've read in other discussions, that one needs to tell the mosquitto addon explicitly to be accesible from the local network.

Where did you read that? From what I know this should work by default: The port 1883 is by default exposed on the host (see Configuration tab of the Mosquitto add-on).

**Comment by SabelOs** [comment]
I've found this post: [https://community.home-assistant.io/t/mosquitto-allow-local-network-access/284833](url). Which I thought migth be the issue, but I had no success when trying the solutions. Also I found this with the same error message: [https://community.home-assistant.io/t/mqtt-error-address-in-use/69113/5](url), although I couldn't find any information on the "plain" setting mentionend here:

> TLDR; set “plain” to false if you are using SSL and an “insecure.conf” file

Also I have the default port configuration in the Addon settings. I tried removing the ssl ports, but with no success:
![image](https://github.com/home-assistant/addons/assets/43387094/ebfe09c6-b5e2-4322-b8f1-ee741f97ee08)

**Comment by agners** [comment]
These are both very old messages, and likely refer to bugs in previous releases or other very specifiic configurations.

By default, the add-on should be reachable without specific configuration. Can you uninstall it again, install it, start it and check if you really can't access?

How do you try to access it? Maybe it is a problem on the other end? Can you try a local MQTT browser or something to try connecting?

**Comment by SabelOs** [comment]
I tried uninstalling and reinstalling the broker, but I still have the same problem.

A bit more information about my setup: I am trying to publish measured data from a Pi Pico to the Home Assistant Mosquitto Broker and then display it in Home Assistant. On the Pico side I use the umqtt.simple library with the following Micro Python code:
`def publish_mqtt_message():`
`    client = MQTTClient("Device-ID", "192.168.0.111", 1883, User, Password)`
`    client.connect()`
`    client.publish(mqtt_topic, mqtt_message)`
`    client.disconnect()`
I know that this code should work because of the documentation and because I tested it with a stand alone version of Mosquitto.

Also I tried to just publish a message with mosquitto_pub from a seperate device containing the same information as the micropython script, which worked. `mosquitto_pub -h 192.168.1.111 -p 1883 -u mqtt-user -P Password -t zigbee2mqtt -i PiZero -m test`

Also I found another problem similar to mine: [https://community.home-assistant.io/t/mqtt-allow-anonymous-login/338345/5](url) and tried the solution in the "Edit2" section with no success. I am not sure what exaclty causes the issue but I think it is with the authentification. Is there an option to disable the user and password requirement? So something similar to "allow_anonymous true".

**Comment by agners** [comment]
> "192.168.0.111"
vs.
> 192.168.1.111

:raised_eyebrow:

**Comment by SabelOs** [comment]
The IP was just a copy error. Thanks nevertheless. I'll ask elsewhere
